.github/workflows/gateway-conformance.yml

```diff
@@ -85,7 +88,28 @@ jobs:
       # 5. Start the kubo-gateway
       - name: Start kubo-gateway
         run: |
-          ./ipfs daemon --offline &
+          ./ipfs daemon &
+          endpoint="http://127.0.0.1:5001/api/v0/version"
+          max_retries=5
+          retry_interval=3
+
+          check_endpoint() {
+            curl -X POST --silent --fail "$endpoint" > /dev/null
+            return $?
+          }
+
+          retries=0
+          while ! check_endpoint; do
+            retries=$((retries+1))
+
+            if [ $retries -ge $max_retries ]; then
+              echo "daemon took too long to start"
+              exit 1
+            fi
+
+            sleep $retry_interval
+          done
+          echo "daemon started and ready to receive API calls"
```

Comment:
Added this here and below. Open to alternatives if people have them, but it seems a little dicey to run the daemon in the background and then start using it later without some kind of a "wait" function.

Maybe we can remove this one if the gateway-conformance binary will do the waiting for us, if so feel free to remove it but we may still need the one for the p2p-proxy.